bots/wallet/bot.js

```diff
@@ -1,13 +1,280 @@
+function calculateFee(n, tx) {
+    var estimatedGas = 21000;
+    if (tx !== null) {
+        estimatedGas = web3.eth.estimateGas(tx);
+    }
+
+    var gasMultiplicator = Math.pow(1.4, n).toFixed(3);
+    return web3.fromWei(web3.eth.gasPrice * gasMultiplicator * estimatedGas, "ether");
+}
+
+function calculateGasPrice(n) {
+    var gasMultiplicator = Math.pow(1.4, n).toFixed(3);
+    return web3.eth.gasPrice * gasMultiplicator;
+}
+
+status.defineSubscription(
+    "calculatedFee",
+    {value: ["sliderValue"], tx: ["transaction"]},
+    function (params) {
+        return calculateFee(params.value, params.tx);
+    }
+);
+
+function getFeeExplaination(n) {
+    return I18n.t('send_explaination') + I18n.t('send_explaination_' + (n + 2));
+}
+
+status.defineSubscription(
+    "feeExplaination",
+    {value: ["sliderValue"]},
+    function(params) {
+        return getFeeExplaination(params.value);
+    }
+)
+
+function amountParameterBox(params, context) {
+    if (!params["bot-db"]) {
+        params["bot-db"] = {};
+    }
+
+    var contactAddress;
+    if (params["bot-db"]["public"] && params["bot-db"]["public"]["recipient"]) {
+        contactAddress = params["bot-db"]["public"]["recipient"]["address"];
+    } else {
+        contactAddress = null;
+    }
+
+    var txData;
+    var amount;
+    try {
+        amount = params.args[1];
+        txData = {
+            to: contactAddress,
+            value: web3.toWei(amount) || 0
+        };
+    } catch (err) {
+        amount = null;
+        txData = {
+            to: contactAddress,
+            value: 0
+        };
+    }
+
+    var sliderValue = params["bot-db"]["sliderValue"] || 0;
+
+    status.setDefaultDb({
+        transaction: txData,
+        calculatedFee: calculateFee(sliderValue, txData),
+        feeExplaination: getFeeExplaination(sliderValue),
+        sliderValue: sliderValue
+    });
+
+    return {
+        title: I18n.t('send_title'),
+        showBack: true,
+        markup: status.components.scrollView(
+            {
+                keyboardShouldPersistTaps: "always"
+            },
+            [status.components.view(
+                {
+                    flex: 1
+                },
+                [
+                    status.components.text(
+                        {
+                            style: {
+                                fontSize: 14,
+                                color: "rgb(147, 155, 161)",
+                                paddingTop: 12,
+                                paddingLeft: 16,
+                                paddingRight: 16,
+                                paddingBottom: 20
+                            }
+                        },
+                        I18n.t('send_specify_amount')
+                    ),
+                    status.components.touchable(
+                        {
+                            onPress: status.components.dispatch([status.events.FOCUS_INPUT, []])
+                        },
```

Comment:
difficult to read code with styles maybe better to move styles in the separate file like in clojure we do?